### PR TITLE
Initial stab at Python 3.8 support

### PR DESCRIPTION
**Pull request by pganssle**
The unusual testing setup makes this very hard to test locally, but this PR is an attempt to get an idea of what exactly needs to be done to support Python 3.8+ in pytype.

### CODE REVIEW

**Comment by martindemello** [comment]
@pganssle thanks for the PR! i'd also like to know more about the problems you had with running the tests; could you open a bug for them?

**Comment by martindemello** [comment]
oh, i see from the comments in the other PR that you are talking about the `tox` and src layout issue, right?

File: pytype/typegraph/cfg.cc [hunk 1]
```diff
@@ -504,7 +504,11 @@ PyTypeObject PyProgram = {
   tp_basicsize : sizeof(PyProgramObj),
   tp_itemsize : 0,
   tp_dealloc : ProgramDealloc,
+#if PY_VERSION_HEX >= 0x03080000 // 3.8
```
Comment (rchen152) [inline pytype/typegraph/cfg.cc:507]: @Solumin FYI

This change looks harmless enough to me but I'm not familiar with this code.

Comment (Solumin) [reply pytype/typegraph/cfg.cc:507]: Yeah, this all looks good: https://www.python.org/dev/peps/pep-0590/#changes-to-the-pytypeobject-struct

Comment (rchen152) [reply pytype/typegraph/cfg.cc:507]: Thanks!

Comment (pganssle) [reply pytype/typegraph/cfg.cc:507]: Ah yeah, FYI I explained each change in the corresponding commit message, see: https://github.com/google/pytype/pull/559/commits/cde5e9c4b0413d8123d64c0d4400efe91e8e64e5

Though I realize that's probably not the best way to communicate it in a PR...

Comment (rchen152) [reply pytype/typegraph/cfg.cc:507]: Ahhh I totally missed that ><

**Comment by pganssle** [comment]
> @pganssle thanks for the PR! i'd also like to know more about the problems you had with running the tests; could you open a bug for them?

Yeah at some point I will open a bug for them, but it's not just `tox` and `src/` environment. These tests require you to have Python 2.7 installed for some reason, they can't be run from a virtual environment, and the outputs are confusing and difficult to interpret.

For example, I don't know what to do next on this PR because the tests fail, but there's no output as to why they failed:

https://travis-ci.org/github/google/pytype/jobs/673128609

When I run them locally (I needed to create a dedicated Python install for this because of the virtual environment problem), it gave me a list of failing tests, and told me to look at the corresponding logs files, all of which were empty.

If you have any tips on how to figure out why these are failing let me know. I was kinda hoping that the travis build would be giving useful information so I could unblock myself, but alas no.

**Comment by rchen152** [comment]
Sorry for letting this sit for so long. Looking back through the comments, I don't think there are any issues that would prevent this from being merged (there's a nit about the placement of a lint directive), so I'm going to go ahead and merge it so the continuation in https://github.com/google/pytype/pull/587 is easier to review.

This PR touches both GitHub-first and google3-first files, so I'm going to merge it directly, then copy the relevant portion of the changes internally and do an export. The commit history will look a little wonky, but hopefully everything will end up in the right state in the end.

**Comment by martindemello** [comment]
addresses #440

**Pull request merged**